app/src/behaviors/behavior_sensor_rotate_common.c

```diff
@@ -5,48 +5,92 @@
 #include <zephyr/kernel.h>
 
 #include <zmk/behavior_queue.h>
+#include <zmk/virtual_key_position.h>
 
 #include "behavior_sensor_rotate_common.h"
 
 LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
 
-int zmk_behavior_sensor_rotate_common_trigger(struct zmk_behavior_binding *binding,
-                                              const struct device *sensor,
-                                              struct zmk_behavior_binding_event event) {
+int zmk_behavior_sensor_rotate_common_data(struct zmk_behavior_binding *binding,
+                                           struct zmk_behavior_binding_event event,
+                                           const struct zmk_sensor_config *sensor_config,
+                                           size_t channel_data_size,
+                                           const struct zmk_sensor_channel_data *channel_data) {
     const struct device *dev = device_get_binding(binding->behavior_dev);
-    const struct behavior_sensor_rotate_config *cfg = dev->config;
+    struct behavior_sensor_rotate_data *data = dev->data;
+
+    const struct sensor_value value = channel_data[0].value;
+    int triggers;
+    int sensor_position = ZMK_SENSOR_POSITION_FROM_VIRTUAL_KEY_POSITION(event.position);
+
+    // Some funky special casing for "old encoder behavior" where ticks where reported in val2 only,
+    // instead of rotational degrees in val1.
+    if (value.val1 == 0) {
+        triggers = value.val2;
+    } else {
+        struct sensor_value remainder = data->remainder[sensor_position];
+
+        remainder.val1 += value.val1;
+        remainder.val2 += value.val2;
+
+        if (remainder.val2 >= 1000000 || remainder.val2 <= 1000000) {
+            remainder.val1 += remainder.val2 / 1000000;
+            remainder.val2 %= 1000000;
+        }
+
+        triggers = remainder.val1 / (360 / sensor_config->triggers_per_rotation);
+        remainder.val1 -= triggers * (360 / sensor_config->triggers_per_rotation);
```

Comment:
As above, consider using modulo to get the remainder.

Reply:
Indeed. Fixed.